File: AspNetMvcUnleashed/0672329980_source/Chapter20/VB/UnleashedBlog/UnleashedBlog/Controllers/BlogController.vb
Imports UnleashedBlog.Attributes

Public Class BlogController
    Inherits Controller

    Private _blogService As BlogServiceBase

    Public Sub New()
        _blogService = New BlogService(Me.ModelState)
    End Sub

    Public Sub New(ByVal blogRepository As BlogRepositoryBase)
        _blogService = New BlogService(Me.ModelState, blogRepository)
    End Sub

    Public Function Index(ByVal page As Integer?) As ActionResult
        Return View(_blogService.ListBlogEntries(page))
    End Function

    <AcceptAjax(), ActionName("Index")> _
    Public Function Index_Ajax(ByVal page As Integer?) As ActionResult
        Return PartialView("BlogEntries", _blogService.ListBlogEntries(page))
    End Function

    Public Function Create() As ActionResult
        Return View()
    End Function

    <AcceptVerbs(HttpVerbs.Post)> _
    Public Function Create(<Bind(Exclude:="Id")> ByVal blogEntryToCreate As BlogEntry) As ActionResult
        If Not _blogService.CreateBlogEntry(blogEntryToCreate) Then
            Return View()
        End If

        Return RedirectToAction("Index")
    End Function
End Class

File: AspNetMvcUnleashed/0672329980_source/Chapter20/CS/UnleashedBlog/UnleashedBlog/Controllers/BlogController.cs
using System.Collections.Generic;
using System.Web.Mvc;
using UnleashedBlog.Models;
using UnleashedBlog.Models.EntityFramework;
using UnleashedBlog.Attributes;

namespace UnleashedBlog.Controllers
{
    public class BlogController : Controller
    {
        private BlogServiceBase _blogService;

        public BlogController()
        {
            _blogService = new BlogService(this.ModelState);
        }

        public BlogController(BlogRepositoryBase blogRepository)
        {
            _blogService = new BlogService(this.ModelState, blogRepository);
        }

        public ActionResult Index(int? page)
        {
            return View(_blogService.ListBlogEntries(page));
        }

        [AcceptAjax]
        [ActionName("Index")]
        public ActionResult Index_Ajax(int? page)
        {
            return PartialView("BlogEntries", _blogService.ListBlogEntries(page));
        }


        public ActionResult Create()
        {
            return View();
        }

        [AcceptVerbs(HttpVerbs.Post)]
        public ActionResult Create([Bind(Exclude="Id")]BlogEntry blogEntryToCreate)
        {
            if (_blogService.CreateBlogEntry(blogEntryToCreate) == false)
                return View();

            return RedirectToAction("Index");
        }
    }
}
